waku/v2/peermanager/peer_connector.go

```diff
@@ -301,21 +256,17 @@ func (c *PeerConnectionStrategy) dialPeers(ctx context.Context) {
 			if c.canDialPeer(pi) {
 				sem <- struct{}{}
 				c.wg.Add(1)
-				go c.dialPeer(pi, sem)
-			} else {
-				continue
+				go c.dialPeer(ctx, pi, sem)
 			}
 		case <-ctx.Done():
 			return
 		}
 	}
 }
 
-func (c *PeerConnectionStrategy) dialPeer(pi peer.AddrInfo, sem chan struct{}) {
+func (c *PeerConnectionStrategy) dialPeer(ctx context.Context, pi peer.AddrInfo, sem chan struct{}) {
 	defer c.wg.Done()
-	c.RLock()
-	ctx, cancel := context.WithTimeout(c.workerCtx, c.dialTimeout)
-	c.RUnlock()
```

Comment:
Ok, but what i was referring to is a case where an app calls start and stop immediately via API's. We have noticed issues because status-go does this during tests. 
I think it would be good to have a test which uses the API to start and stop in quick succession.